environment.py

```diff
@@ -16,6 +20,43 @@
 logger = logging.getLogger(__name__)
 
 
+def download_model_files(repo_url: str, repo_dir: str, local_path: str) -> None:
+    """Download or update model files (XML + meshes) from a GitHub repository.
+
+    :param repo_url: URL of the GitHub repository
+    :param repo_dir: Directory within the repository containing the model files
+    :param local_path: Local path where files should be saved
+    """
+    target_path = Path(local_path) / repo_dir
+
+    # Check if the target directory already exists
+    if target_path.exists():
+        logger.info(f"Model files are already present in {target_path}. Skipping download.")
+        return
+
+    # Create a temporary directory for cloning
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
this is not the right way to do this because it will make a temp directory every time you run the script - instead, you should have it set up so that you make some consistent directory, for example in `Path(__file__).parent.resolve() / "data"` and clone there if the data is missing there

Reply:
If i clone into the repo it removes all other folders in the environments/ folder so I fixed it this way by allowing it to cache multiple environments. It doesn't make a temp directory everytime the user would run the script because I have the 
```
# Check if the target directory already exists
    if target_path.exists():
        logger.info(f"Model files are already present in {target_path}. Skipping download.")
        return
  ```
  
  check. Only way this fails if the user edits the xml and reruns (but for some reason doesn't want to keep their edits, but I'm supposing the user may want to edit the xml to add collisions / new poses and still be able to rerun the script while keeping their changes. Another failure mode would be if I'm updating the Github repo as a user wants to get the newest updates, but they can just fix by deleting the folder.

Reply:
This is actually a good feature because if just clone all the time, the diff would always update and override the user's changes which would be kind of bad.

Reply:
ohhhhhhhh sorry, i am dumb, you did this correctly :) nice job